python/pyspark/sql/tests.py

```diff
@@ -2367,6 +2380,157 @@ def range_frame_match():
 
         importlib.reload(window)
 
+
+class TypesTest(unittest.TestCase):
+
+    def test_verify_type_ok_nullable(self):
+        for obj, data_type in [
+                (None, IntegerType()),
+                (None, FloatType()),
+                (None, StringType()),
+                (None, StructType([]))]:
```

Comment:
Meaning remove the None from the tuples and just loop over the Types?  I like it a little better as is since the tuples are basically `_verify_type`'s args but am fine with either.  Let me know which you prefer and I can change the code.